src/MongoDB.Driver/Linq/Linq3Implementation/Misc/ConvertHelper.cs

```diff
@@ -37,6 +36,21 @@ public static Expression RemoveConvertToMongoQueryable(Expression expression)
             throw new ExpressionNotSupportedException(expression);
         }
 
+        public static Expression RemoveEnumConvert(Expression expression)
+        {
+            if (expression.NodeType == ExpressionType.Convert)
+            {
+                var convertExpression = (UnaryExpression)expression;
+                var sourceType = convertExpression.Operand.Type;
+                if (sourceType.IsEnum())
```

Comment:
Let's be a bit more precise about when it's OK to remove the `Convert`

    var targetType = convertExpression.Type;
    if (sourceType.IsEnum() && targetType == Enum.GetUnderlyingType(sourceType))

Reply:
Done.